### ISSUE TITLE
AzureChinaCloud Restic URL ERROR

### ISSUE THREAD

**Issue by XiaFanGit** [vmware-tanzu/velero#3386]
when I use velero + restic to backup aks cluster. the restic repo is <blobname>.blob.core.windows.net.
however, in AzureChinaCLoud it should to be <blobname>.blob.core.chinacloudapi.cn

* The output of the following commands

```
cat credentials-velero 
AZURE_STORAGE_ACCOUNT_ACCESS_KEY=<accesskeys>
AZURE_CLOUD_NAME=AzureChinaCloud
```

```
velero install \
    --provider azure \
    --plugins velero/velero-plugin-for-microsoft-azure:v1.1.0 \
    --bucket velero \
    --secret-file ./credentials-velero \
    --backup-location-config resourceGroup=<RG>,storageAccount=<account_name>,storageAccountKeyEnvVar=AZURE_STORAGE_ACCOUNT_ACCESS_KEY,subscriptionId=<subscriptionId>\
    --use-volume-snapshots=false \
    --use-restic
```

```
kubectl logs deployment/velero -n velero
time="2021-02-01T10:20:01Z" level=info msg="No backup locations were ready to be verified" controller=backupstoragelocation logSource="pkg/controller/backupstoragelocation_controller.go:120"
```

```
velero backup logs kube-logging|grep error
time="2021-02-01T10:13:36Z" level=error msg="Error backing up item" backup=velero/kube-logging error="restic repository is not ready: error running command=restic init --repo=azure:velero:/restic/kube-logging --password-file=/tmp/velero-restic-credentials-kube-logging392714186 --cache-dir=/scratch/.cache/restic, stdout=, stderr=Fatal: create repository at azure:velero:/restic/kube-logging failed: container.CreateIfNotExists: Put https://velerostaging.blob.core.windows.net/velero?restype=container: dial tcp: lookup velerostaging.blob.core.windows.net on 10.0.0.10:53: no such host\n\n: exit status 1" error.file="/go/src/github.com/vmware-tanzu/velero/pkg/restic/repository_ensurer.go:144" error.function="github.com/vmware-tanzu/velero/pkg/restic.(*repositoryEnsurer).EnsureRepo" logSource="pkg/backup/backup.go:431" name=nutstore-log-es-master-nodes-0
```

**Environment:**

- Velero version (use `velero version`): 
```
velero version
Client:
        Version: v1.5.3
        Git commit: 123109a3bcac11dbb6783d2758207bac0d0817cb
Server:
        Version: v1.5.3
```
- Velero features (use `velero client config get features`): none

- Kubernetes version (use `kubectl version`): 1.18.14

- Cloud provider or hardware configuration: AKS

**Comment by zubron** [comment]
This is an open issue with restic as it does not provide a way to configure the domain to use: https://github.com/restic/restic/issues/2468

The suggested workaround there is to use the `rclone` backend for restic however this is is not currently supported in Velero. We have an open issue to support more backends: #1178.

**Comment by XiaFanGit** [comment]
@zubron Hi, Thanks a lot for your answer. I compiled restic manually and it work fine. Thx

**Comment by dsu-igeek** [comment]
We'll leave this open until we can integrate the fixes to Restic.

**Comment by reasonerjt** [comment]
This is not a high priority for Restic as we are moving away from it.  
@Lyndon-Li please make sure in v1.10 kopia integration this can be supported.

**Comment by Lyndon-Li** [comment]
This problem is fixed for Kopia path in PR #5142

**Comment by Lyndon-Li** [comment]
I am closing this issue since we have addressed it in Kopia path.
